test/unit/test_utils.js

```diff
@@ -122,73 +122,91 @@ function createIdFactory(pageIndex) {
   return page._localIdFactory;
 }
 
-function createTemporaryNodeServer() {
-  assert(isNodeJS, "Should only be used in Node.js environments.");
-
-  const fs = process.getBuiltinModule("fs"),
-    http = process.getBuiltinModule("http");
-  function isAcceptablePath(requestUrl) {
-    try {
-      // Reject unnormalized paths, to protect against path traversal attacks.
-      const url = new URL(requestUrl, "https://localhost/");
-      return url.pathname === requestUrl;
-    } catch {
-      return false;
+// Some tests rely on special behavior from webserver.mjs. When loaded in the
+// browser, the page is already served from WebServer. When running from
+// Node.js, that is not the case. This helper starts the WebServer if needed,
+// and offers a mechanism to resolve the URL in a uniform way.
+class TestPdfsServer {
+  static #webServer;
+
+  static #startCount = 0;
+
+  static async ensureStarted() {
+    if (this.#startCount++) {
+      // Already started before. E.g. from another beforeAll call.
+      return;
```

Comment:
If there's two back-to-back calls to `ensureStarted` then the second one may resolve *before* the server is actually available, which seems like it could cause trouble, since loading/initializing is async below.